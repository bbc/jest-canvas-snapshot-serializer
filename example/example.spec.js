const canvasSnapshotSerializer = require("../src/index");

expect.addSnapshotSerializer(canvasSnapshotSerializer);

describe("my test suite", () => {
  it("works", () => {
    const canvas = createCanvas("#000", "#F00");
    expect(canvas).toMatchSnapshot();
  });

  it("works for canvas wrapped by other elements", () => {
    const canvas = createCanvas("#000", "#0F0");
    const root = document.createElement("div");
    root.appendChild(canvas);

    expect(root).toMatchSnapshot();
  });

  function createCanvas(color, textColor) {
    const canvas = document.createElement("canvas");
    const context = canvas.getContext("2d");

    canvas.setAttribute("height", "200");
    canvas.setAttribute("height", "200");

    context.fillStyle = color;
    context.beginPath();
    context.arc(100, 100, 75, 0, 2 * Math.PI);
    context.fill();

    context.font = "24px Helvetica";
    context.fillStyle = textColor;
    context.fillText("Canvas", 50, 130);

    return canvas;
  }
});
