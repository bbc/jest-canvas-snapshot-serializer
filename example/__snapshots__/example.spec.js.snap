// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my test suite works 1`] = `
<canvas
  data-snapshot-image="99e2e6c9af9c059f1946964e4dbe5914d22d952261d6f26f0f013ed1a2f7f5677cb3643108bd25a90382cba24bf0fe84a1719d043b7063cbff8359b172ecb791"
  height="200"
  width="200"
/>
`;

exports[`my test suite works for canvas wrapped by other elements 1`] = `
<div>
  <canvas
  data-snapshot-image="f02c73bff2ff9dad8908b9ca2fa63c02d6d92cd91d46f87e539bb3c79ba7cbcbdb1251a3db132598fb4be3eed46c5150e2c70395f3089d1f8a860c3836645b0d"
  height="200"
  width="200"
/>
</div>
`;
